v2/robotmk/src/bin/agent.rs

```diff
@@ -0,0 +1,139 @@
+use camino::Utf8PathBuf;
+use clap::Parser;
+use serde::{Deserialize, Serialize};
+use std::env::{var, VarError};
+use std::fs::read_to_string;
+use std::io;
+use std::path::Path;
+use walkdir::{DirEntry, WalkDir};
+
+#[derive(Deserialize)]
+pub struct Config {
+    pub results_directory: Utf8PathBuf,
+}
+
+#[derive(Parser)]
+#[command(about = "Robotmk agent plugin.")]
+struct Args {
+    /// Configuration file path.
+    #[clap(name = "CONFIG_PATH")]
+    pub config_path: Option<Utf8PathBuf>,
+}
+
+#[derive(Serialize)]
+pub struct ConfigError {
+    config_reading_error: String,
+}
+
+#[derive(Serialize)]
+pub struct ConfigFileContent {
+    config_file_content: String,
+}
+
+fn determine_config_path(arg: Option<Utf8PathBuf>) -> Result<Utf8PathBuf, String> {
+    if let Some(p) = arg {
+        return Ok(p);
+    }
+    let config_dir = match var("MK_CONFDIR") {
+        Ok(path) => path,
+        Err(VarError::NotPresent) => "C:\\ProgramData\\checkmk\\agent\\config".to_string(),
+        Err(VarError::NotUnicode(_path)) => return Err("CONFIG_PATH is not utf-8.".into()),
+    };
+    let mut config_dir = Utf8PathBuf::from(config_dir);
+    config_dir.push("robotmk.json");
+    Ok(config_dir)
+}
+
+fn report_config_error(message: String) {
+    let config_error = serde_json::to_string(&ConfigError {
+        config_reading_error: message,
+    })
+    .unwrap();
+    println!("{config_error}");
+}
+
+fn report_config_content(content: String) {
+    let config_content = serde_json::to_string(&ConfigFileContent {
+        config_file_content: content,
+    })
+    .unwrap();
+    println!("{config_content}");
+}
+
+fn print_or_ignore(entry: Result<DirEntry, walkdir::Error>, stdout: &mut impl io::Write) {
+    if let Ok(dir) = entry {
+        if dir.file_type().is_file() {
+            if let Ok(raw) = read_to_string(dir.path()) {
+                writeln!(stdout, "{raw}").unwrap();
+            }
+        }
+    }
+}
+
+fn walk(results_directory: &Path, stdout: &mut impl io::Write) {
+    for entry in WalkDir::new(results_directory)
+        .sort_by_file_name()
+        .into_iter()
```

Comment:
Since we are (correctly) ignoring errors in `print_or_ignore`, you could directly transform into an iterator which contains `DirEntry` by discarding any errors:
```
.filter_map(|dir_entry_result| dir_entry_result.ok())
```
Then you would have one nesting level less in `print_or_ignore`.